src/utils.rs

```diff
@@ -752,16 +721,48 @@ pub fn open_image(img_location: &Path) -> Result<FrameCollection> {
             let d = DynamicImage::ImageRgb8(x);
             col.add_still(d.to_rgba8());
         }
-        #[cfg(feature = "jpgxl")]
         "jxl" => {
-            use jpegxl_rs::decoder_builder;
-            use jpegxl_rs::image::ToDynamic;
-            let sample = std::fs::read(&img_location)?;
-            let decoder = decoder_builder().build()?;
-            let img = decoder
-                .decode_to_image(&sample)?
-                .context("Can't decode image from jpgxl")?;
-            col.add_still(img.into_rgba8());
+            let mut image = JxlImage::open(img_location).map_err(|e| anyhow!("{e}"))?;
+            info!("{:?}", image.image_header());
+            let mut renderer = image.renderer();
+            loop {
+                let result = renderer
+                    .render_next_frame()
+                    .map_err(|e| anyhow!("{e}"))
+                    .context("Can't render JXL")?;
+                match result {
+                    RenderResult::Done(render) => {
+                        let i = render.image();
+
+                        // JXL with alpha
+                        if render.extra_channels().len() == 1 {
```

Comment:
Thank you for integrating jxl-oxide
```suggestion
                        if renderer.pixel_format().has_alpha() {
```
Because there are many types of extra_channels, such as alpha, depth, spot color, selection mask, black, thermal, etc.